### PR TITLE
find_type function doesn't recognize exported incident types

### PR DESCRIPTION
**Pull request by rshunim**
## Status
- [x] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-1494)

## Must have
- [ ] Tests
- [ ] Documentation
- [ ] [VS Code Extension](https://docs.gitlab.com/ee/ci/yaml/#tags)
  - [ ] Functionality implemented in the extension - <u>\<link-to-pr-here\></u>
  - [ ] N/A - Functionality cannot be implemented in the extension because <u>\<add-reason-here\></u>

### CODE REVIEW

File: demisto_sdk/commands/common/tools.py [hunk 1]
```diff
@@ -2125,7 +2129,8 @@ def get_file_displayed_name(file_path):
                        FileType.INDICATOR_FIELD, FileType.LAYOUTS_CONTAINER, FileType.PRE_PROCESS_RULES,
                        FileType.DASHBOARD, FileType.WIDGET,
                        FileType.REPORT, FileType.JOB, FileType.WIZARD]:
-        return get_json(file_path).get('name')
+        res = get_json(file_path)
+        return res.get('name') if isinstance(res, dict) else res[0].get('name')
```
Comment (dorschw) [inline demisto_sdk/commands/common/tools.py:2133]: if it's a list, is it always of size 1? 

Perhaps add a check: if res is not a dict, and ((not list) or (list of len >1)) log a warning that it's an unexpected value and the reason

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +11,7 @@
 * **validate** will no longer fail on playbooks calling subplaybooks that have a higher `fromVersion` value, if  calling the subplaybook has `skipifunavailable=True`.
 * Fixed an issue where relative paths were not accessed correctly.
 * Running any `demisto-sdk` command in a folder with a `.env` file will load it, temporarily overriding existing environment variables.
+* Fixed an issue where **find_type** function didn't recognize exported incident types.
```
Comment (dorschw) [inline CHANGELOG.md:14]: the bold part is usually the SDK command users use, I suppose it's mostly format that uses `find_type`?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (dorschw) [inline CHANGELOG.md:1]:
```suggestion
```

File: CHANGELOG.md (hunk 2 above)
Comment (rshunim) [reply CHANGELOG.md:14]: I'm not sure, also validate, upload, secrets, generic, lint and other

File: demisto_sdk/commands/common/tools.py (hunk 1 above)
Comment (rshunim) [reply demisto_sdk/commands/common/tools.py:2133]: Let's talk about it

File: demisto_sdk/tests/test_files/serverIncidentType.json [hunk 4]
```diff
@@ -0,0 +1,33 @@
+[
```
Comment (dorschw) [inline demisto_sdk/tests/test_files/serverIncidentType.json:1]: please rename the file `IncidentType__raw_downdloaded.json`

File: demisto_sdk/tests/test_files/contentIncidentType.json [hunk 5]
```diff
@@ -0,0 +1,31 @@
+{
```
Comment (dorschw) [inline demisto_sdk/tests/test_files/contentIncidentType.json:1]: please rename to `IncidentType.json`

File: demisto_sdk/tests/constants_test.py [hunk 6]
```diff
@@ -122,6 +122,8 @@
 INVALID_INTEGRATION_YML_5 = f"{GIT_ROOT}/demisto_sdk/tests/test_files/integration-invalid-yml5.yml"
 VALID_REPUTATION_FILE = f"{GIT_ROOT}/demisto_sdk/tests/test_files/reputation-cidr-valid.json"
 INVALID_REPUTATION_FILE = f"{GIT_ROOT}/demisto_sdk/tests/test_files/reputation-cidr-invalid.json"
+SERVER_VALID_INCIDENT_TYPE_FILE = f"{GIT_ROOT}/demisto_sdk/tests/test_files/serverIncidentType.json"
+CONTENT_VALID_INCIDENT_TYPE_FILE = f"{GIT_ROOT}/demisto_sdk/tests/test_files/contentIncidentType.json"
```
Comment (dorschw) [inline demisto_sdk/tests/constants_test.py:126]:
```suggestion
VALID_INCIDENT_TYPE_FILE__RAW_DOWNLOAD = f"{GIT_ROOT}/demisto_sdk/tests/test_files/IncidentType__raw_downdloaded.json"
VALID_INCIDENT_TYPE_FILE = f"{GIT_ROOT}/demisto_sdk/tests/test_files/IncidentType.json"
```

File: demisto_sdk/commands/common/tests/tools_test.py [hunk 7]
```diff
@@ -1041,6 +1044,27 @@ def test_get_file_displayed_name__image(repo):
...
+
+    When
+    - Running get_file_displayed_name.
+
+    Then:
+    - Ensure the returned name is the incident type name.
+    """
+
+    display_name = get_file_displayed_name(input_path)
+    assert display_name == expected_name
```
Comment (dorschw) [inline demisto_sdk/commands/common/tests/tools_test.py:1065]:
```suggestion
    assert get_file_displayed_name(input_path) == expected_name
```

File: demisto_sdk/commands/common/tools.py [hunk 8]
```diff
@@ -1279,7 +1279,11 @@ def get_dict_from_file(path: str,
             if path.endswith('.yml'):
                 return get_yaml(path, cache_clear=clear_cache), 'yml'
             elif path.endswith('.json'):
-                return get_json(path, cache_clear=clear_cache), 'json'
+                res = get_json(path, cache_clear=clear_cache)
```
Comment (dorschw) [inline demisto_sdk/commands/common/tools.py:1282]: please update the docstrings, mentioning the method returns the dict or first item if the file contents are a list with a single dictionary

**Review by dorschw — APPROVED** [review]

File: CHANGELOG.md [hunk 9]
```diff
@@ -11,6 +11,7 @@
 * Fixed an issue where relative paths were not accessed correctly.
 * Running any `demisto-sdk` command in a folder with a `.env` file will load it, temporarily overriding existing environment variables.
 * Fixed an issue where **validate** did not properly detect deleted files.
+* Fixed an issue where **find_type** function didn't recognize exported incident types.
```
Comment (dorschw) [inline CHANGELOG.md:14]:
```suggestion
* Fixed an issue where *find_type* didn't recognize exported incident types.
```

**Pull request merged**